packages/lib-runtime/src/store/usePluginInfo.ts

```diff
@@ -0,0 +1,25 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import type { PluginInfoEntry, PluginConsumer } from '../types/store';
+import { PluginEventType } from '../types/store';
+import { usePluginSubscription } from './usePluginSubscription';
+
+/**
+ * React hook for consuming current information about plugins.
+ *
+ * This hook re-renders the component whenever the plugin information changes.
+ *
+ * The hook's result is guaranteed to be referentially stable across re-renders.
+ */
+export const usePluginInfo = (): PluginInfoEntry[] => {
+  const getData = React.useCallback(
+    (pluginConsumer: PluginConsumer) => pluginConsumer.getPluginInfo(),
+    [],
+  );
+
+  const eventTypes = React.useMemo(() => [PluginEventType.PluginInfoChanged], []);
```

Comment:
Any reason this isn't just a file constant?

Reply:
In general, React hooks should be sensitive (responsive) to changes to their arguments. That's because the hook's arguments typically have a direct impact on the hook's result.

Let's take the `usePluginSubscription` hook as an example. It has 3 arguments: `getData` (function), `isSameData` (function) and `eventTypes` (array of strings).

If we call this hook like so:

```ts
const infoEntries = usePluginSubscription(
  (pluginConsumer) => pluginConsumer.getPluginInfo(),
  (prevData, nextData) => _.isEqual(prevData, nextData),
  [PluginEventType.PluginInfoChanged],
);
```

we're effectively creating brand new `getData`, `isSameData` and `eventTypes` argument objects _every time the hook gets called_. And because of that, the `usePluginSubscription` hook will un- and re-subscribe to `PluginStore` every time we call it that way, which is highly inefficient and just plain wrong.

What we want to do is to ensure that all `usePluginSubscription` arguments change (in terms of the referenced object) only when they need to. In our case, there's nothing that impacts these arguments, so we memoize their values using an empty dependency array `[]` which tells React the value never changes.

```
const getData = React.useCallback(
  (pluginConsumer: PluginConsumer) => pluginConsumer.getPluginInfo(),
  [], // no dependencies, getData reference never changes
);
```

From React docs on [`useCallback`](https://reactjs.org/docs/hooks-reference.html#usecallback)

> `useCallback(fn, deps)` is equivalent to `useMemo(() => fn, deps)`

Reply:
@vojtechszocs what @andrewballantyne is asking here is why not define this as a constant _outside_ the component instead of creating the new array each time and using `useMemo`

Reply:
Aww, I came here to say that @christianvogt -- stealing all my thunder! 😛

Reply:
Sorry, seems like I managed to completely misunderstand @andrewballantyne's comment!

Agreed, will do the change.